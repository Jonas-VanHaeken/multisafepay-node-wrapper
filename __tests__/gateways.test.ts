import axios from 'axios';
import Gateways from '../src/resources/gateways';
import allGatewaysResponse from '../__stubs__/allGateways.stub.json';
import gatewaybyIDResponse from '../__stubs__/gatewayByID.stub.json';
import MockAdapter from 'axios-mock-adapter';

const mock = new MockAdapter(axios);

describe('gateways', () => {
  let gateways: any;
  beforeEach(() => {
    gateways = new Gateways(axios.create());
  });

  it('check endpoint', () => {
    expect(Gateways.endpoint).toBe('gateways');
  });

  it('get all gateways', async () => {
    mock.onGet('/gateways').reply(200, allGatewaysResponse);

    const resp = await gateways.get();

    expect(resp).toMatchSnapshot();
  });

  it('get gateways by ID ex:IDEAL', async () => {
    mock.onGet('/gateways/IDEAL').reply(200, gatewaybyIDResponse);

    const resp = await gateways.get('IDEAL');

    expect(resp).toMatchSnapshot();
  });

  it('get gateways by wrong ID', async () => {
    const error = {
      error_code: 1023,
      error_info: 'No gateway (payment method) available',
    };
    mock.onGet('/gateways/hello').reply(404, error);

    const resp = await gateways.get('hello');

    expect(resp).toEqual(error);
  });
});
