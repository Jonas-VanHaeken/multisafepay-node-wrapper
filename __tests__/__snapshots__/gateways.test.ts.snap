// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gateways get all gateways 1`] = `
Object {
  "data": Array [
    Object {
      "description": "iDEAL",
      "id": "IDEAL",
    },
    Object {
      "description": "Bancontact",
      "id": "MISTERCASH",
    },
    Object {
      "description": "Visa",
      "id": "VISA",
    },
    Object {
      "description": "Maestro",
      "id": "MAESTRO",
    },
    Object {
      "description": "PayPal",
      "id": "PAYPAL",
    },
    Object {
      "description": "DotPay",
      "id": "DOTPAY",
    },
    Object {
      "description": "Giropay",
      "id": "GIROPAY",
    },
    Object {
      "description": "EPS",
      "id": "EPS",
    },
    Object {
      "description": "MasterCard",
      "id": "MASTERCARD",
    },
    Object {
      "description": "Wire Transfer",
      "id": "BANKTRANS",
    },
    Object {
      "description": "American Express",
      "id": "AMEX",
    },
    Object {
      "description": "Direct Debit",
      "id": "DIRDEB",
    },
    Object {
      "description": "SOFORT Banking",
      "id": "DIRECTBANK",
    },
    Object {
      "description": "ING HomePay",
      "id": "INGHOME",
    },
    Object {
      "description": "Pay After Delivery",
      "id": "PAYAFTER",
    },
    Object {
      "description": "JCB",
      "id": "JCB",
    },
    Object {
      "description": "Santander",
      "id": "SANTANDER",
    },
    Object {
      "description": "Klarna",
      "id": "KLARNA",
    },
    Object {
      "description": "Trustly",
      "id": "TRUSTLY",
    },
    Object {
      "description": "Alipay",
      "id": "ALIPAY",
    },
    Object {
      "description": "Einvoice",
      "id": "EINVOICE",
    },
    Object {
      "description": "Belfius",
      "id": "BELFIUS",
    },
    Object {
      "description": "Pay In Advance",
      "id": "PAYINADV",
    },
    Object {
      "description": "iDEAL QR",
      "id": "IDEALQR",
    },
    Object {
      "description": "Pay After Delivery B2B",
      "id": "PAYAFTB2B",
    },
    Object {
      "description": "KBC",
      "id": "KBC",
    },
    Object {
      "description": "AfterPay",
      "id": "AFTERPAY",
    },
    Object {
      "description": "fietsenbon",
      "id": "FIETSENBON",
      "type": "coupon",
    },
  ],
  "success": true,
}
`;

exports[`gateways get gateways by ID ex:IDEAL 1`] = `
Object {
  "message": Object {
    "data": Object {
      "description": "Pay After Delivery",
      "id": "PAYAFTER",
    },
    "success": true,
  },
}
`;
